pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -695,21 +697,45 @@ func (qjm *XController) getAppWrapperCompletionStatus(caw *arbv1.AppWrapper) arb
 			}
 			unstruct.Object = blob.(map[string]interface{}) //set object to the content of the blob after Unmarshalling
 			name := ""
+			namespace := ""
 			if md, ok := unstruct.Object["metadata"]; ok {
 				metadata := md.(map[string]interface{})
 				if objectName, ok := metadata["name"]; ok {
 					name = objectName.(string)
 				}
+				if objectName, ok := metadata["namespace"]; ok {
```

Comment:
Please add a test case for non-namedspaced test case to ensure a nil here would not panic.  Thanks!

Reply:
if objectNamespace, ok := metadata["namespace"]; ok {
					if objectNamespace != nil {
						namespace = objectNamespace.(string)
					}
				}